prompt.go

```diff
@@ -269,6 +283,8 @@ func (p *Prompt) setUp() {
 }
 
 func (p *Prompt) tearDown() {
-	debug.AssertNoError(p.in.TearDown())
+	if p.inNotTearDown {
```

Comment:
How about this change, instead of introducing `p.inNotTearDown`? `p.exitor` called twice but I think it's no problem at almost situations. It might be better to warn it in godoc comment of `OptionSetExitCheckerOnInput`.

```suggestion
	if p.exitor != nil && p.exitor(e.input) {
```

Reply:
The previous comment (line 81) means no need for `inNotTearDown`